runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java

```diff
@@ -36,22 +48,100 @@
 )
 public class GroundItemsPlugin extends Plugin
 {
-	@Inject
-	private ConfigManager configManager;
+	// Splitting the items in the config on commas
+	private static final Splitter COMMA_SPLITTER = Splitter.on(Pattern.compile("\\s*,\\s*"));
 
 	@Inject
 	private GroundItemsOverlay overlay;
 
+	@Inject
+	private GroundItemsConfig config;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Getter(AccessLevel.PACKAGE)
+	private List<String> hiddenItems;
+
+	@Getter(AccessLevel.PACKAGE)
+	private List<String> highlightedItems;
+
+	@Getter(AccessLevel.PACKAGE)
+	private LoadingCache<String, Boolean> highlightedItemAssociations;
+
+	@Getter(AccessLevel.PACKAGE)
+	private LoadingCache<String, Boolean> hiddenItemAssociations;
+
 	@Provides
 	GroundItemsConfig provideConfig(ConfigManager configManager)
 	{
 		return configManager.getConfig(GroundItemsConfig.class);
 	}
 
+	@Override
+	protected void startUp()
+	{
+		updateConfig();
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		hiddenItems = null;
+		highlightedItems = null;
+		hiddenItemAssociations.cleanUp();
+		hiddenItemAssociations = null;
+		highlightedItemAssociations.cleanUp();
+		highlightedItemAssociations = null;
+	}
+
 	@Override
 	public Overlay getOverlay()
 	{
 		return overlay;
 	}
 
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("grounditems"))
+		{
+			return;
+		}
+
+		updateConfig();
+	}
+
+	private void updateConfig()
+	{
+		// gets the hidden items from the text box in the config
+		hiddenItems = ImmutableList.copyOf(COMMA_SPLITTER
+			.splitToList(config.getHiddenItems().toLowerCase().trim()));
+
+		// gets the highlighted items from the text box in the config
+		highlightedItems = ImmutableList.copyOf(COMMA_SPLITTER
+			.splitToList(config.getHighlightItems().toLowerCase().trim()));
+
+		// cleanup previous association caches
+		if (hiddenItemAssociations != null)
+		{
+			hiddenItemAssociations.cleanUp();
+		}
+
+		if (highlightedItemAssociations != null)
+		{
+			highlightedItemAssociations.cleanUp();
+		}
+
+		// create new association caches
+		hiddenItemAssociations = CacheBuilder.newBuilder()
```

Comment:
Create these in startUp and then invalidate them here

Reply:
Can't do, they are created from parsed highlighted and hidden item lists.